skimage/measure/fit.py

```diff
@@ -610,19 +612,14 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
     trials : int
         Number of trials.
     """
+    if probability == 0:
+        return 0
     if n_inliers == 0:
         return np.inf
-
-    if probability == 1:
-        return np.inf
-
-    if n_inliers == n_samples:
-        return 1
-
-    nom = math.log(1 - probability)
-    denom = math.log(1 - (n_inliers / n_samples) ** min_samples)
-
-    return int(np.ceil(nom / denom))
+    inlier_ratio = n_inliers / n_samples
+    nom = max(_EPSILON, 1 - probability)
+    denom = max(_EPSILON, 1 - inlier_ratio ** min_samples)
+    return abs(float(np.ceil(np.log(nom) / np.log(denom))))
```

Comment:
I think that the use of `abs` is not necessary: both `nom` and `denom` are in `[_EPSILON, 1]`, so both `log` values are negative and theire ratio is positive. Right?

Reply:
I agree on it. I'll remove `abs`.

Reply:
```suggestion
    return np.ceil(np.log(nom) / np.log(denom))
```